src/components/TodoItem.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoContext } from '../TodoContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [query, setQuery] = useState(todo.title);
+  const editingInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editingInput.current) {
+      editingInput.current.focus();
+    }
+  }, [isEditing]);
+
+  const deleteTodofromStorage = (todoId: number) => {
+    const updatedTodos = todos.filter(currentTodo => currentTodo.id !== todoId);
+
+    setTodos(updatedTodos);
+  };
+
+  const updateTitle = (id: number, newTitle: string) => {
+    const updatedTodos = todos.map(currentTodo => {
+      if (currentTodo.id === id) {
+        return {
+          ...currentTodo,
+          title: newTitle,
+        };
+      }
+
+      return currentTodo;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  const submitUpdateTitle = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (query && query !== todo.title) {
+      updateTitle(todo.id, query);
+      setIsEditing(false);
+    }
+
+    if (!query?.length) {
+      deleteTodofromStorage(todo.id);
+    }
+
+    setIsEditing(false);
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case 'Enter':
+        if (query) {
+          updateTitle(todo.id, query);
+        }
+
+        setIsEditing(false);
```

Comment:
you did it in the `submitUpdateTitle` function
and generally - it's a redundant case - it will always be handled by `submitUpdateTitle`